nvm.sh

```diff
@@ -13,13 +13,7 @@
 NVM_SCRIPT_SOURCE="$_"
 
 nvm_echo() {
-  command printf %s\\n "$*" 2>/dev/null || {
-    nvm_echo() {
-      # shellcheck disable=SC1001
-      \printf %s\\n "$*" # on zsh, `command printf` sometimes fails
```

Comment:
wow, is the implication here that this failure is solely based on using a var named "path" in `nvm_find_up`??

If so, then I think that shellcheck should warn for variables named `path` when set to zsh mode - that would have caught this issue much sooner. ~~I'll file one :-)~~ Update: I filed https://github.com/koalaman/shellcheck/issues/1058, but turns out shellcheck doesn't care about zsh.

Reply:
The backslash makes alias no effect and is useless in scripting.

Thus this snippet works as "use shell builtin when command fails".

Note: I just noticed that `command printf` is shell builtin in bash, too. (Don't know why.)